images/dex/tests/02-runs.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+if [[ "${IMAGE_NAME}" == "" ]]; then
+	echo "Must set IMAGE_NAME environment variable. Exiting."
+	exit 1
+fi
+
+PORT=5556
+
+URL="http://127.0.0.1:${PORT}"
+docker run --rm -d -p $PORT:$PORT "${IMAGE_NAME}" dex serve /etc/dex/config.docker.yaml
+
+# Maximum number of attempts to check the health check endpoint
+MAX_ATTEMPTS=30
+
+# Delay (in seconds) between attempts
+DELAY=5
+
+attempt=0
+until $(curl --output /dev/null --silent --head --fail "$URL/dex/healthz"); do
+	if [ $attempt -eq $MAX_ATTEMPTS ]; then
+		echo "Max attempts reached, exiting..."
+		exit 1
+	fi
+
+	printf '.'
+	attempt=$(($attempt + 1))
+
+	# Delay before next attempt
+	sleep $DELAY
+done
```

Comment:
I liked the way how you handle the resiliency, but we can achieve similar way with `curl`. Just pass the following flags: `--max-time 60 --retry 3 --retry-max-time 120`. We can tackle this in one-liner, if you liked the idea.